zerostash/src/commands/zfs/zfs_commit.rs

```diff
@@ -0,0 +1,49 @@
+//! `zfs commit` subcommand
+
+use infinitree::Infinitree;
+use zerostash_files::{Files, Snapshot};
+
+use crate::prelude::*;
+
+#[derive(Command, Debug)]
+pub struct ZfsCommit {
+    #[clap(flatten)]
+    stash: StashArgs,
+
+    /// Commit message to include in the changeset
+    #[clap(short = 'm', long)]
+    message: Option<String>,
+
+    /// Snapshot name
+    #[clap(long)]
+    snapshot: String,
+}
+
+#[async_trait]
+impl AsyncRunnable for ZfsCommit {
+    /// Start the application.
+    async fn run(&self) {
+        let mut stash = self.stash.open();
+        stash.load(stash.index().snapshots()).unwrap();
+
+        add_snapshot(&stash, self.snapshot.clone());
+
+        stash
+            .commit(self.message.clone())
```

Comment:
You can use `Option::take` to avoid cloning.

Reply:
Is it ok to change to `&mut self`? It causes a lot of errors I would have to fix.

Reply:
I think because `Runnable` uses `&self` I would need to clone cmd, but that doesn't sound like its worth it to me. Am I missing something?